File: Homeworks/C#1/04.ConsoleInputOutput/06.QuadraticEquation/QuadraticEquation.cs
using System;

class QuadraticEquation
{
    static void Main()
    {
         Console.Write("Please, enter intex a: ");
        double a = double.Parse(Console.ReadLine());
        Console.Write("Please, enter intex b: ");
        double b = double.Parse(Console.ReadLine());
        Console.Write("Please, enter intex c: ");
        double c = double.Parse(Console.ReadLine());

        double D = Math.Pow(b, 2) - 4 * a * c;
        if (D < 0)
        {
            Console.WriteLine("No real roots.");
        }
        else if (D == 0)
        {
            double x = -b / (2 * a);
            Console.WriteLine("x1 = x2 = {0}", x);
        }
        else
        {
            double x1 = (-b + Math.Sqrt(D)) / (2 * a);
            double x2 = (-b - Math.Sqrt(D)) / (2 * a);

            Console.WriteLine("x1 = {0}\tx2 = {1}", x1, x2);
        }
    }
}

File: Homeworks/JavaScriptFundamentals/03.ConditionalStatements/06.QuadraticEquation/QuadraticEquation.js
/*
 Write a script that enters the coefficients a, b and c
 of a quadratic equation
 a*x2 + b*x + c = 0
 and calculates and prints its real roots. Note that
 quadratic equations may have 0, 1 or 2 real roots.
 */

function quadraticEquation(form) {
    var a = eval(form.a.value);
    var b = eval(form.b.value);
    var c = eval(form.c.value);

    var x1 = -b / 2 / a + Math.pow(Math.pow(b, 2) - 4 * a * c, 0.5) / 2 / a;
    var x2 = -b / 2 / a - Math.pow(Math.pow(b, 2) - 4 * a * c, 0.5) / 2 / a;

    form.x1.value = x1;
    form.x2.value = x2;

    if (form.x1.value == "NaN") form.x1.value = "Imaginary!";
    if (form.x2.value == "NaN") form.x2.value = "Imaginary!";
}